### PR TITLE
Proper output filename

### PR DESCRIPTION
**Pull request by bidzyyys**
https://golemproject.atlassian.net/secure/RapidBoard.jspa?rapidView=4&modal=detail&selectedIssue=CGI-71

### CODE REVIEW

File: TranscodingDemo/Server/views.py [hunk 1]
```diff
@@ -83,6 +83,11 @@ def get(self, request, ident):
...
             path = Result.get_path(job=job, identifier=ident)
-            return FileResponse(open(path, 'rb'), as_attachment=True)
+            return FileResponse(open(path, 'rb'), as_attachment=True,
+                                filename=Logic.get_final_output_filename(path,
+                                                                         request.session["TranscodingParams"][
+                                                                             "VideoName"]))
         except (FileNotFoundError, UnauthorizedUser) as e:
             return JsonResponse(status=http.HTTPStatus.BAD_REQUEST, data=str(e), safe=False)
+        except Exception as e:
+            return JsonResponse(status=http.HTTPStatus.INTERNAL_SERVER_ERROR, data={"ErrorMessage": repr(e)})
```
Comment (weaselix) [inline TranscodingDemo/Server/views.py]: Don't reveal our internal exception message to the client.

File: TranscodingDemo/Server/applogic.py [hunk 2]
```diff
@@ -327,3 +324,9 @@ def transcoding_progress_logic_get(request):
         cxt = Logic.get_context_transcoding_progress(request.session["SrcVideoParams"],
                                                      request.session["TranscodingParams"], job)
         return None, cxt
+
+    @staticmethod
+    def get_final_output_filename(requested, src):
+        _, ext = os.path.splitext(requested)
+        filename, _ = os.path.splitext(src)
+        return "{}{}".format(filename, ext)
```
Comment (weaselix) [inline TranscodingDemo/Server/applogic.py]: Don't you need a `.` here?

Comment (bidzyyys) [reply TranscodingDemo/Server/applogic.py]: I've made the same mistake, look at the python console output
```
>>> os.path.splitext("test.mp4")
('test', '.mp4')
```

Comment (weaselix) [reply TranscodingDemo/Server/applogic.py]: ok

**Pull request merged**